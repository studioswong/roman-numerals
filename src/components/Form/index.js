import React, { Component } from 'react';
import styled from 'styled-components';
import CalculateValue from '../../utils/CalculateValue';

export default class Form extends React.Component {
    constructor(props) {
      super(props);
      this.state = {rValue: '', iValue:0};

      this.handleChange = this.handleChange.bind(this);
      this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
      this.setState({rValue: event.target.value});
    }

    handleSubmit(event) {
      event.preventDefault();
      this.setState({iValue: CalculateValue(this.state.rValue)});
    }

    render() {
      return (
        <div >
            <form onSubmit={this.handleSubmit}>
              <label>
                Enter Roman Numerals:<br/>
                <input type="text" value={this.state.value} onChange={this.handleChange} />
              </label>
              <input type="submit" value="Submit" />
            </form>
            <p> This equates to:</p>
            <div>{this.state.iValue}</div>
        </div>
      );
    }
  }

