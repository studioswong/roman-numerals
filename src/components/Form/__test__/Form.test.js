/**
 * @jest-environment node
 */

import React from 'react';
import { shallow } from 'enzyme';
import Form from '../';

describe('Form UI', () => {
    it('should render', () => {
        const wrapper = shallow(<Form />);
        expect(wrapper).toMatchSnapshot();
    });

});