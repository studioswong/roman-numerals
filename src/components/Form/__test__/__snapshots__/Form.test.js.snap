// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form UI should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
        >
          <label>
            Enter Roman Numerals:
            <br />
            <input
              onChange={[Function]}
              type="text"
              value={undefined}
            />
          </label>
          <input
            type="submit"
            value="Submit"
          />
        </form>,
        <p>
           This equates to:
        </p>,
        <div>
          0
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Enter Roman Numerals:
              <br />
              <input
                onChange={[Function]}
                type="text"
                value={undefined}
              />
            </label>,
            <input
              type="submit"
              value="Submit"
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Enter Roman Numerals:",
                <br />,
                <input
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Enter Roman Numerals:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onChange": [Function],
                  "type": "text",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
              "value": "Submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " This equates to:",
        },
        "ref": null,
        "rendered": " This equates to:",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
        },
        "ref": null,
        "rendered": 0,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
          >
            <label>
              Enter Roman Numerals:
              <br />
              <input
                onChange={[Function]}
                type="text"
                value={undefined}
              />
            </label>
            <input
              type="submit"
              value="Submit"
            />
          </form>,
          <p>
             This equates to:
          </p>,
          <div>
            0
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Enter Roman Numerals:
                <br />
                <input
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </label>,
              <input
                type="submit"
                value="Submit"
              />,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Enter Roman Numerals:",
                  <br />,
                  <input
                    onChange={[Function]}
                    type="text"
                    value={undefined}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Enter Roman Numerals:",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "onChange": [Function],
                    "type": "text",
                    "value": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "Submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " This equates to:",
          },
          "ref": null,
          "rendered": " This equates to:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
          },
          "ref": null,
          "rendered": 0,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
